src/toolbox/utils/color/color.spec.ts

```diff
@@ -1,52 +1,103 @@
+  describe("constructor", () => {
     const tests = [
-      [255,0,0,0],
-      [0,255,0,0],
-      [0,255,0,1],
-      [0,255,0,.25],
-      [30,25,234,.25],
+      [255, 0, 0, 0],
+      [0, 255, 0, 0],
+      [0, 255, 0, 1],
+      [0, 255, 0, 0.25],
+      [30, 25, 234, 0.25]
     ];
-    tests.forEach((testValues) => {
-      it(
-        `should return ${JSON.stringify(testValues)} from toNumbers when
-         constructed with ${JSON.stringify(testValues)}`,
-        () => {
-          expect(
-            areArrayValuesEqual(
-              new Color(...testValues).toNumbers(), testValues)
-          ).to.equal(true);
-        }
-      );
+    tests.forEach(testValues => {
+      it(`should return ${JSON.stringify(testValues)} from toNumbers when
+         constructed with ${JSON.stringify(testValues)}`, () => {
+        expect(
+          areArrayValuesEqual(new Color(...testValues).toNumbers(), testValues)
+        ).to.equal(true);
+      });
     });
   });
 
-  describe('fromStyleString (static)', () => {
+  describe("fromStyleString (static)", () => {
     const tests = [
-      ['red', [255,0,0,1]],
-      ['hotpink', [255,105,180,1]],
-      ['#ff69b4', [255,105,180,1]],
-      ['rgb(255,105,180)', [255,105,180,1]],
-      ['rgba(255,105,180,1)', [255,105,180,1]],
-      ['rgba(255,105,180,0.5)', [255,105,180,.5]],
-      ['rgba(255,105,180,.5)', [255,105,180,.5]],
-      ['#FFF', [255,255,255,1]],
+      ["red", [255, 0, 0, 1]],
+      ["hotpink", [255, 105, 180, 1]],
+      ["#ff69b4", [255, 105, 180, 1]],
+      ["rgb(255,105,180)", [255, 105, 180, 1]],
+      ["rgba(255,105,180,1)", [255, 105, 180, 1]],
+      ["rgba(255,105,180,0.5)", [255, 105, 180, 0.5]],
+      ["rgba(255,105,180,.5)", [255, 105, 180, 0.5]],
+      ["#FFF", [255, 255, 255, 1]]
     ];
     tests.forEach(([testInput, expectedResult]) => {
-      it(
-        `should return ${JSON.stringify(expectedResult)} from toNumbers when
-         constructed with Color.fromStyleString(${JSON.stringify(testInput)})`,
-        () => {
-          const result = Color.fromStyleString(<string>testInput).toNumbers();
-          expect(result).to.have.ordered.members(<number[]>expectedResult);
-          expect(result.length).to.equal((<number[]>expectedResult).length)
-        }
-      );
+      it(`should return ${JSON.stringify(expectedResult)} from toNumbers when
+         constructed with Color.fromStyleString(${JSON.stringify(
+           testInput
+         )})`, () => {
+        const result = Color.fromStyleString(<string>testInput).toNumbers();
+        expect(result).to.have.ordered.members(<number[]>expectedResult);
+        expect(result.length).to.equal((<number[]>expectedResult).length);
+      });
+    });
+  });
+
+  describe("fromHex", () => {
+    const tests = [
+      ["#FFFFFF", [255, 255, 255, 1]],
+      ["#000000", [0, 0, 0, 1]],
+      ["#812dd3", [129, 45, 211, 1]],
+      ["#F2F", [255, 34, 255, 1]],
+      ["FFFFFF", [255, 255, 255, 1]]
```

Comment:
Does this work without the hash prefix if it is a 3-character color code like "DdD"?

Reply:
Yes

Reply:
Cool :) let's make sure we have a test to back that up so that it doesn't change accidentally if someone updates the code.